alibi_detect/cd/classifier.py

```diff
@@ -142,6 +143,9 @@ def __init__(
         """
         super().__init__()
 
+        # Get args/kwargs to set config later
+        inputs = locals().copy()
```

Comment:
will this duplicate the memory during init even if it's later gc'd?

Reply:
P.s. IMO the above complexity is a symptom of the way we set detector backends i.e. `self._detector = MMDDriftTF(...)`. It is slightly odd how `MMDDrift` is not a detector itself, but only a container with a detector as an attribute. 

I've been discussing with @mauicv how we might want to refactor this, and will add some ideas to https://github.com/SeldonIO/alibi-detect/issues/512.